### PR TITLE
Update Nodemailer production transport options for testing

### LINKED ISSUE #149: Modify Nodemailer module for testing in production environment
## Description

@humphd on Slack
 >we should modify your code a bit so that it can accept OAuth2 info or username/password in production, and for now we'll use the latter

Update the Nodemailer module production code to accept username/password for the initial testing of the current notification functionality.

### PR DESCRIPTION
**Pull request by SerpentBytes**
## Description
This PR resolves #149 by updating the `production` transport options with the following settings:
- host: `smtp.office365.com`
- port: `587`
- auth (accepts username, pass)
- requireTLS: `true  // forces STARTTLS`

### CODE REVIEW

File: app/lib/notifications.server.ts
Comment (humphd) [inline app/lib/notifications.server.ts:23]: This will always be `587` in production, let's not bother with a variable to configure it.

File: app/lib/notifications.server.ts
Comment (humphd) [inline app/lib/notifications.server.ts:18]: I would improve this error message: tell them which variables are actually missing, so it's easier to debug startup crashes.

File: .env.example
Comment (humphd) [inline .env.example:17]: We probably don't need this, see my comment later.

**Review by humphd — CHANGES_REQUESTED** [review]
Nice, this is looking good.  A few small things.

**Comment by SerpentBytes** [comment]
> Nice, this is looking good.  A few small things.

I was thinking of expanding this PR by implementing tests that would work with MailHog. That way, we would know the current configuration of nodemailer is working as intended.

**Review by dadolhay — APPROVED** [review]

**Review by sfrunza13 — DISMISSED** [review]

**Comment by humphd** [comment]
@SerpentBytes looks like this is good to merge.

**Comment by SerpentBytes** [comment]
> @SerpentBytes looks like this is good to merge.

merging is blocked despite have 2 approvals.

**Review by shawnyu5 — APPROVED** [review]

**Review by humphd — APPROVED** [review]

**Comment by humphd** [comment]
Should be good now, I'm not sure why it blocked on my review.

**Pull request merged**
